src/main/kotlin/com/exactpro/th2/sim/template/rule/KotlinFIXRule.kt

```diff
@@ -44,15 +44,17 @@ class KotlinFIXRule(field: Map<String, Any?>) : MessageCompareRule() {
         private val TrdMatchId = AtomicInteger(0)
 
         private val buyOrdersAndIds: Queue<Pair<ParsedMessage, Int>> = LinkedList()
-        private val sellOrders: Queue<Pair<ParsedMessage, Int>> = LinkedList()
+        private val sellOrdersAndIds: Queue<Pair<ParsedMessage, Int>> = LinkedList()
 
         fun reset() {
             orderId.set(0)
             execId.set(0)
             TrdMatchId.set(0)
 
-            buyOrdersAndIds.clear()
-            sellOrders.clear()
+            synchronized(KotlinFIXRule::class.java) {
```

Comment:
1) could you please create a separate lock object for locking
2) we must use locking for each state operations (read/write)

Reply:
It's not necessary to use locking for adding new order to queue because it can't affect processing previously added orders (but we need to use concurrent version of queue). We only need locking to ensure three consequent orders from two queues will be extracted at once, if more orders will be added to queue tail at the same time is not a problem. We also don't need to use locking with id counters as since using of AtomicInteger is enough to avoid duplicated ids.